bin/FvwmCommand.in

```diff
@@ -0,0 +1,143 @@
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+
+    # Connect the socket to the "port" where the server is listening
+    # FvwmMFL is setting this in environment, so we can really treat
+    # absence of the environment FVWMMFL_SOCKET as error, except when
+    # explicit -f <sockname> has been specified by the user.
+    if not fvwm_socket:
+        fvwm_socket = str(os.environ.get('FVWMMFL_SOCKET'))
+    try:
+        if verbose:
+            print ("Trying to determine if", fvwm_socket, "exists as MFL socket ...")
+        mode = os.stat(fvwm_socket).st_mode
+        if stat.S_ISSOCK(mode):
+            server_address = fvwm_socket
+            if verbose:
+                print ("Using", fvwm_socket, "as MFL socket.")
+    except FileNotFoundError:
+        print ("Cannot find FvwmMFL socket " + fvwm_socket + ".")
+        sys.exit(1)
+
+    if verbose:
+        print('Connecting to {}'.format(server_address))
+    try:
+        sock.connect(server_address)
+    except socket.error as msg:
+        print(msg)
+        sys.exit(1)
+
+    try:
+        if verbose:
+            print ("Sending data to MFL socket", server_address)
+        # Send data, either multiple commands from pipe, arguments
+        if read_from_stdin:
+            for stdline in sys.stdin:
+                message = stdline.encode()
+                sock.sendall(message)
+        else:
+            message = sys.argv[-1].encode()
+            if sys.argv[0] is not sys.argv[-1]:
+                sock.sendall(message)
+        if verbose:
+            print('Sent {!r}'.format(message) + " to FvwmMFL")
+    
+        amount_received = 0
+        data = sock.recv(32768)
+        # amount_received += len(data)
+        # print (amount_received)
+        if verbose:
+            print('Received {!r}'.format(data))
+        if info:
+            # Weak dependency only if -i was used.
+            import json
+            jparsed = json.loads(data.decode('utf-8'))
+            print (json.dumps(jparsed, sort_keys=True, indent=4))
+
+    # Close unix domain socket
+    finally:
+        if verbose:
+            print('Closing connection to socket', server_address)
+        sock.close()
+
+def usage():
+    sockname = str(os.environ.get('FVWMMFL_SOCKET'))
+    text = """Usage: FvwmCommand [OPTION] [COMMAND]...
+Send commands to fvwm3 via FvwmMFL
+
+  -c                  read and send commands from stdin
+                      The COMMAND if any is ignored.
+  -f <file name>      use unix socket <file name> to connect to FvwmMFL
+  -i <info level>     0 - default
+                      >0 - print FvwmMFL feedback
+  -d                  print detailed diagnostic messages
+  -v                  print version number
+
+Default unix socket name is:"""
+    print (text, sockname)
+
+def main():
+    # Defaults for mflclnt function
+    verbose = False
+    info = False
+    read_from_stdin = False
+    fvwm_socket = False
+    versioninfo = "__FVWM3_VERSIONINFO__"
```

Comment:
With autotools, and how we use it with fvwm, we have some predefined variables which are figured out at `./configure` time and the substituted across the code base.  One of these is `VERSION` and `VERSIONINFO`.  So here, you just need to do:

    versioninfo=@VERSION@-@VERSIONINFO@

... and autotools will substitute the appropriate values.

This simplifies the Makefile as well -- I've left a comment there.